controller/aks-cluster-config-handler.go

```diff
@@ -704,14 +704,13 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 			updateNodePool := false
 			upstreamNodePool, ok := upstreamNodePools[npName]
 			if ok {
-				if to.Bool(np.EnableAutoScaling) {
+				if to.Bool(np.EnableAutoScaling) && !to.Bool(upstreamNodePool.EnableAutoScaling) {
 					if to.Int32(np.Count) != to.Int32(upstreamNodePool.Count) {
 						return config, fmt.Errorf("can't update node count when autoscaling is enabled in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
 					}
-					if !to.Bool(upstreamNodePool.EnableAutoScaling) {
-						logrus.Infof("Enable autoscaling in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
-						updateNodePool = true
-					}
+					logrus.Infof("Enable autoscaling in node pool [%s] for cluster [%s]", to.String(np.Name), config.Spec.ClusterName)
+					updateNodePool = true
+				} else if to.Bool(np.EnableAutoScaling) && to.Bool(upstreamNodePool.EnableAutoScaling) {
```

Comment:
Using `else if` here means that you can't change both `EnableAutoScaling` and `MinCount` or `MaxCount` at the same time, the change to `MinCount` would be ignored by the first case.

Reply:
It seems that `MinCount` and `MaxCount` can be changed regardless of the value of `EnableAutoScaling`. If this is correct, I think refactoring to make this section easier to comprehend would be valuable.

Reply:
Current logic is OK. We have to recognize situation when:
- `EnableAutoScalling` is enabled on downstream and disabled on upstream, then `NodeCount` can't be changed, `MinCount` and `MaxCount` will be updated anyway
- `EnableAutoScalling` is enabled on downstream and upstream, then only  `MinCount` and `MaxCount` can be updated, `NodeCount` node count will be changed by AKS auto-scaler
- `EnableAutoScalling` is disabled in downstream and still enabled on upstream, then `MinCount` and `MaxCount` should be `nil`

This is Azure API logic